similarity/metrics.py

```diff
@@ -0,0 +1,290 @@
+    description = 'GFCCs'
+    category = 'timbre'
+    path = "data->'lowlevel'->'gfcc'->'mean'"
+    indices = range(1, 13)
+
+
+class WeightedGfccsMetric(WeightedNormalizedLowLevelMetric, GfccsMetric):
+    name = 'gfccsw'
+    description = 'GFCCs (weighted)'
+
+
+class CircularMetric(LowLevelMetric):
+    def length(self):
+        return 2
+
+    def transform(self, data):
+        """Wrap the value that around circle with overlaps on integers"""
+        value = data * 2 * np.pi
+        return np.array([np.cos(value), np.sin(value)])
+
+
+KEYS_CIRCLE = ['C', 'G', 'D', 'A', 'E', 'B', 'F#', 'C#', 'G#', 'D#', 'A#', 'F']
+KEYS_MAP = {KEYS_CIRCLE[i]: float(i) / 12 for i in range(12)}
+SCALES_MAP = {'major': 0.0, 'minor': -3.0 / 12}
+
+
+class KeyMetric(CircularMetric):
+    name = 'key'
+    path = "data->'tonal'"
+    category = 'rhythm'
+    description = 'Key/Scale'
+
+    def transform(self, data):
+        try:
+            key_value = KEYS_MAP[data['key_key']]
+            key_value += SCALES_MAP[data['key_scale']]
+            return super(KeyMetric, self).transform(key_value)
+        except KeyError:
+            raise ValueError('Invalid key/scale values: {}, {}'.format(data['key_key'], data['key_scale']))
+
+
+class LogCircularMetric(CircularMetric):
+    def transform(self, data):
+        if not data:
+            raise ValueError('Invalid data value: {}'.format(data))
+        return super(LogCircularMetric, self).transform(np.log2(data))
+
+
+class BpmMetric(LogCircularMetric):
+    name = 'bpm'
+    description = 'BPM'
+    category = 'rhythm'
+    path = "data->'rhythm'->'bpm'"
+
+
+class OnsetRateMetric(LogCircularMetric):
+    name = 'onsetrate'
+    description = 'OnsetRate'
+    category = 'rhythm'
+    path = "data->'rhythm'->'onset_rate'"
+
+
+class HighLevelMetric(BaseMetric):
+    category = 'high-level'
+
+    def get_data(self, id):
+        data = db.data.get_highlevel_models(id)
+        return data
+
+
+class BinaryCollectiveMetric(HighLevelMetric):
+    models = None
+
+    def transform(self, data):
+        if not data:
+            raise ValueError('Invalid data value: {}'.format(data))
+
+        vector = []
+        try:
+            for model_id in self.models:
+                if str(model_id) in data.keys():
+                    model_data = data[str(model_id)]['all']
+                    assert len(model_data) == 2
+                    for key, value in model_data.items():
+                        if not key.startswith('not'):
+                            vector.append(value)
+                            break
+                else:
+                    vector.append(None)
+            return vector
+        except KeyError:
+            raise ValueError('Invalid data value: {}'.format(data))
+
+    def length(self):
+        return len(self.models)
+
+
+class MoodsMetric(BinaryCollectiveMetric):
+    name = 'moods'
+    description = 'Moods'
+    models = [
+        11,  # mood_happy
```

Comment:
My idea was always to use the name of the model to identify it. is there any reason that you can see why we can't select from the model table by the name instead of the id?